### DISCUSSION TITLE
Centralized Audit Log of all developer and admin actions

### DISCUSSION THREAD

**Discussion by kdecks** [RetroAchievements/RAWeb#653] [Ideas]
Including:
- Changing/adding game page info and images
- Changing/adding/removing leaderboards
- Closing Tickets
- Changing/Adding rich presence
- Adding/Demoting/Editing achievements
- Renaming a Game
- Removing leaderboard scores
- Unlinking/Linking MD5s
- Deleting a forum thread
- Deleting comments (admin)
- Editing forum comments (admin)
- Banning (admin)
- Untracking (admin)
- Manually added achievements (admin)

what else?

**Comment by kdecks** [comment]
With the audit log in place giving the permission for Admin to be able to delete achievements from players would no longer be a security concern and this permission could be granted.

**Comment by kdecks** [comment]
I would add, after an audit exists:
- allow admin to **remove** achievements from players.

Normal times for removal:
- manual awarding mistakes
- cheating
- helping out players seeking help
- easy removal of achievements that were earned from buggy achievements

**Comment by meleu** [comment]
> - easy removal of achievements that were earned from buggy achievements

yeah! an easy way to remove a known buggy cheevo from all players that earned it should be handy for admins.
